### PR TITLE
Fix data ingestion from Amazon S3 bucket

### PR DESCRIPTION
**Pull request by KKcorps**
This PR fixes the issue #5835 of ingesting data from S3.

The fix involves the following changes 

- Return full paths along with scheme in listFiles functions instead of path after the bucket. 
This change is needed because the output of the listFiles in passed to isDirectory function which throws Exception since the path doesn't contain the bucket

- Take only the path instead of the full URI in segment generation job runner while creating a local file.
The segment generator throws the error `inputFileURI doesn't start with file://` on creating the local file. This small change of taking `inputFileURI.getPath()` instead of `inputFileURI` fixes this issue.

The changes have been verified by running a standalone job to ingest as well as upload data in S3.

### CODE REVIEW

**Comment by KKcorps** [comment]
@fx19880617 @npawar

File: pinot-plugins/pinot-file-system/pinot-s3/src/test/java/org/apache/pinot/plugin/filesystem/S3PinotFSTest.java [hunk 1]
```diff
@@ -28,12 +28,16 @@
 import java.util.Arrays;
 import java.util.List;
 import org.apache.commons.io.IOUtils;
+import org.apache.pinot.spi.env.PinotConfiguration;
```
Comment (fx19880617) [inline pinot-plugins/pinot-file-system/pinot-s3/src/test/java/org/apache/pinot/plugin/filesystem/S3PinotFSTest.java:31]: do we need those imports?

Comment (KKcorps) [reply pinot-plugins/pinot-file-system/pinot-s3/src/test/java/org/apache/pinot/plugin/filesystem/S3PinotFSTest.java]: Removed unused imports.

File: pinot-plugins/pinot-file-system/pinot-s3/src/main/java/org/apache/pinot/plugin/filesystem/S3PinotFS.java [hunk 2]
```diff
@@ -391,7 +393,11 @@ public long length(URI fileUri)
       listObjectsV2Response.contents().stream().forEach(object -> {
         //Only add files and not directories
         if (!object.key().equals(fileUri.getPath()) && !object.key().endsWith(DELIMITER)) {
-          builder.add(object.key());
+          String fileKey = object.key();
+          if(fileKey.startsWith(DELIMITER)){
```
[the remark this answers is not in the document]
Comment (KKcorps) [reply pinot-plugins/pinot-file-system/pinot-s3/src/main/java/org/apache/pinot/plugin/filesystem/S3PinotFS.java]: does it look fine now?

**Pull request merged**